mirrord/config/src/feature/env/mapper.rs

```diff
@@ -0,0 +1,114 @@
+use std::collections::HashMap;
+
+use fancy_regex::Regex;
+use tracing::Level;
+
+#[derive(Debug)]
+pub struct EnvVarsRemapper {
+    mapping: Vec<(Regex, String)>,
+    env_vars: HashMap<String, String>,
+}
+
+impl EnvVarsRemapper {
+    #[tracing::instrument(level = Level::TRACE, ret)]
+    pub fn new(mapping: HashMap<String, String>, env_vars: HashMap<String, String>) -> Self {
+        let mapping = mapping
+            .into_iter()
+            .map(|(pattern, value)| {
+                (
+                    Regex::new(&pattern).expect("Building env vars mapping regex failed"),
+                    value,
+                )
+            })
+            .collect();
+
+        EnvVarsRemapper { mapping, env_vars }
+    }
+
+    #[tracing::instrument(level = Level::TRACE, ret)]
+    pub fn remapped(self) -> HashMap<String, String> {
+        let Self { mapping, env_vars } = self;
+
+        env_vars
+            .into_iter()
+            .fold(HashMap::with_capacity(32), |mut acc, (key, value)| {
+                match mapping
+                    .iter()
+                    .find(|(k, _)| k.is_match(&key).ok().is_some_and(|matched| matched))
+                    .map(|(_, to_value)| to_value.clone())
+                {
+                    Some(to_value) => acc.insert(key, to_value),
+                    None => acc.insert(key, value),
+                };
+
+                acc
+            })
```

Comment:
This is a bit weird :eyes: 
Can't we do
```rust
env_vars
  .iter_mut()
  .for_each(|(name, value)| {
    ....
  });

env_vars
```

Also, I don't see capture groups support :eyes:

Reply:
Oops.

Changed it to `captures`.

Reply:
Hmm I think we run into some annoying borrowing issues.

Reply:
> Changed it to `captures`.

Sorry, I was wrong. I thought capture groups support meant that the user can use capture groups in value (in other words, value is a template that can refer to the captured portions of the key). `is_match` is fine here

Reply:
Change to `is_match` thanks to your other suggestion.